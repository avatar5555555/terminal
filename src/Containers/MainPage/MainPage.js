// @flow

import React, { Component } from 'react'
import Grid from '@material-ui/core/Grid'
import CircularProgress from '@material-ui/core/CircularProgress'

import OperatorCard from 'src/Components/OperatorCard'
import type { MainPageProps } from 'src/types'

class MainPage extends Component<MainPageProps> {
  componentDidMount() {
    this.props.fetchList()
  }

  render() {
    const { list, isLoading, classes } = this.props

    return (
      <Grid container spacing={16} alignItems="center" direction="column">
        {isLoading ? (
          <CircularProgress
            id="spinner"
            size={70}
            className={classes.spinner}
          />
        ) : (
          list.map(item => <OperatorCard key={item.name} {...item} />)
        )}
      </Grid>
    )
  }
}

export default MainPage
