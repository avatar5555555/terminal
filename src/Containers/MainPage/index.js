// @flow

import React, { Component } from 'react'
import Grid from '@material-ui/core/Grid'
import { connect } from 'react-redux'
import CircularProgress from '@material-ui/core/CircularProgress'
import { compose } from 'redux'
import { withStyles } from '@material-ui/core/styles'

import { getOperatorsList, getIsLoading } from 'src/store/selectors'
import { operatorsListReadRequest } from 'src/store/actions'
import OperatorCard from 'src/Components/OperatorCard'

type Item = {
  src: string,
  name: string
}

type Props = {
  fetchList: Function,
  list: Array<Item>,
  isLoading: boolean,
  classes: { spinner: string }
}

const styles = theme => ({
  spinner: {
    marginTop: theme.spacing.unit * 10
  }
})

class MainPage extends Component<Props> {
  componentDidMount() {
    this.props.fetchList()
  }

  render() {
    const { list, isLoading, classes } = this.props

    return (
      <Grid container spacing={16} alignItems="center" direction="column">
        {isLoading ? (
          <CircularProgress size={70} className={classes.spinner} />
        ) : (
          list.map(item => <OperatorCard key={item.name} {...item} />)
        )}
      </Grid>
    )
  }
}

const mapState = state => ({
  list: getOperatorsList(state),
  isLoading: getIsLoading(state)
})

const bindActions = dispatch => ({
  fetchList: () => dispatch(operatorsListReadRequest())
})

export default compose(
  withStyles(styles),
  connect(
    mapState,
    bindActions
  )
)(MainPage)
