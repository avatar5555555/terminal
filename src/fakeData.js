const fakeData = [
  { name: 'mts', src: '/Images/mts.png' },
  { name: 'beeline', src: '/Images/beeline.png' },
  { name: 'megafon', src: '/Images/megafon.png' },
  { name: 'other', src: '/Images/other.png' }
]

export default fakeData
