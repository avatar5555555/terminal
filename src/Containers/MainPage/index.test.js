import React from 'react'
import { shallow } from 'enzyme'

import MainPage from './MainPage'

import { styles, mapState, bindActions } from '.'

import fakeData from 'src/fakeData'

const props = {
  fetchList: jest.fn(),
  list: fakeData,
  isLoading: false,
  classes: { spinner: 'string' }
}

describe('MainPage', () => {
  const wrap = (props = {}) => shallow(<MainPage {...props} />)
  it('renders', () => expect(wrap(props)).toMatchSnapshot())

  it('renders spinner if data is loading', () => {
    const wrapper = wrap({ ...props, isLoading: true })
    const spinner = wrapper.dive().find('#spinner')

    expect(spinner).toBeDefined()
  })
})

describe('MainPage functions', () => {
  it('styles return object with where unit multiplied at 10', () => {
    const obj = styles({ spacing: { unit: 5 } })

    expect(obj.spinner.marginTop).toBe(50)
  })

  it('bindActions return obj with fn', () => {
    const mockDispatch = jest.fn()
    const actions = bindActions(mockDispatch)
    actions.fetchList()

    expect(mockDispatch.mock.calls.length).toBe(1)
  })

  it('mapState return new object', () => {
    const mappedProps = mapState({ list: fakeData, isLoading: false })

    expect(mappedProps.list).toEqual(fakeData)
  })
})
