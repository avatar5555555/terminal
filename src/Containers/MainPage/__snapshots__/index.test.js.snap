// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainPage
    classes={
        Object {
            "spinner": "string",
          }
    }
    fetchList={[Function]}
    isLoading={false}
    list={
        Array [
            Object {
              "name": "mts",
              "src": "/Images/mts.png",
            },
            Object {
              "name": "beeline",
              "src": "/Images/beeline.png",
            },
            Object {
              "name": "megafon",
              "src": "/Images/megafon.png",
            },
            Object {
              "name": "other",
              "src": "/Images/other.png",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "alignItems": "center",
      "children": Array [
        <WithStyles(OperatorCard)
          name="mts"
          src="/Images/mts.png"
/>,
        <WithStyles(OperatorCard)
          name="beeline"
          src="/Images/beeline.png"
/>,
        <WithStyles(OperatorCard)
          name="megafon"
          src="/Images/megafon.png"
/>,
        <WithStyles(OperatorCard)
          name="other"
          src="/Images/other.png"
/>,
      ],
      "container": true,
      "direction": "column",
      "spacing": 16,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "mts",
        "nodeType": "class",
        "props": Object {
          "name": "mts",
          "src": "/Images/mts.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "beeline",
        "nodeType": "class",
        "props": Object {
          "name": "beeline",
          "src": "/Images/beeline.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "megafon",
        "nodeType": "class",
        "props": Object {
          "name": "megafon",
          "src": "/Images/megafon.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "other",
        "nodeType": "class",
        "props": Object {
          "name": "other",
          "src": "/Images/other.png",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "alignItems": "center",
        "children": Array [
          <WithStyles(OperatorCard)
            name="mts"
            src="/Images/mts.png"
/>,
          <WithStyles(OperatorCard)
            name="beeline"
            src="/Images/beeline.png"
/>,
          <WithStyles(OperatorCard)
            name="megafon"
            src="/Images/megafon.png"
/>,
          <WithStyles(OperatorCard)
            name="other"
            src="/Images/other.png"
/>,
        ],
        "container": true,
        "direction": "column",
        "spacing": 16,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "mts",
          "nodeType": "class",
          "props": Object {
            "name": "mts",
            "src": "/Images/mts.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "beeline",
          "nodeType": "class",
          "props": Object {
            "name": "beeline",
            "src": "/Images/beeline.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "megafon",
          "nodeType": "class",
          "props": Object {
            "name": "megafon",
            "src": "/Images/megafon.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "other",
          "nodeType": "class",
          "props": Object {
            "name": "other",
            "src": "/Images/other.png",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
